src/web3/BaseClient.ts

```diff
@@ -99,7 +118,7 @@ export class BaseClient {
    *
    * @returns An array of IProvider containing all the private providers.
    */
-  protected getPrivateProviders(): Array<IProvider> {
+  public getPrivateProviders(): Array<IProvider> {
```

Comment:
I don't think making this function public is the best choice. Protected functions protect us from breaking changes.
I suggest 2 ideas:
1. Through the constructor
2. Test subclass: A subclass specifically for testing, which exposes the protected functions.